nixos/modules/virtualisation/gce-images.nix

```diff
@@ -5,5 +5,13 @@ let self = {
   "17.03" = "gs://nixos-cloud-images/nixos-image-17.03.1082.4aab5c5798-x86_64-linux.raw.tar.gz";
   "18.03" = "gs://nixos-cloud-images/nixos-image-18.03.132536.fdb5ba4cdf9-x86_64-linux.raw.tar.gz";
   "18.09" = "gs://nixos-cloud-images/nixos-image-18.09.1228.a4c4cbb613c-x86_64-linux.raw.tar.gz";
-  latest = self."18.09";
+
+  # This format will be handled by the upcoming NixOPS 2.0 release.
+  # The old images based on a GS object are deprecated.
```

Comment:
@AmineChikhaoui does this mean one has to use NixOps to use these images? Or can one derive the GS path somehow?

Reply:
I think it should be usable outside of NixOPS but I haven't tried that yet.
Looks like `gcloud compute images list --project nixos-cloud` doesn't show the nixos-* images as it might need `compute.images.list` to all users. I'll see if that's ok to add for better discoverability.

Reply:
@AmineChikhaoui any update about this? :pray:

Reply:
Although I guess even if they can't be listed, they're available as sources under `"gs://${BUCKET_NAME}/$img_name"`, right? It's just that one will have to "reverse-engineer" the image name with `BUCKET_NAME=nixos-cloud` and `img_name=nixos-image-20-09-3531-3858fbc08e6-x86-64-linux`.

Correct?

Reply:
@asymmetric correct, I've just added the list permission to all users of the bucket so that it's possible to list objects. I'd have to look at the IAM docs of google to figure out how to make image listing possible as well.

Reply:
@AmineChikhaoui not sure I understand: you want to make listing of objects in the bucket possible for everyone? If so, I think [this](https://cloud.google.com/storage/docs/access-control/making-data-public#buckets) is how oyu do that.

Both the `nixos-images` and the `nixos-cloud-images` allow that, so if you have access to those, you could also look how they do it.

Reply:
@asymmetric that I already did. I want this to work for all users and show the nixos-* images:
```
gcloud compute images list --project nixos-cloud
```

Reply:
@AmineChikhaoui is [this] what you're looking for?

Reply:
@AmineChikhaoui I think we've seen this before per https://github.com/nix-community/nixops-gce/pull/9#issuecomment-648062625

The only fix was to use the full URL as described per : https://github.com/nix-community/nixops-gce/pull/9/commits/fe1c19e6582d50b433de3dd08f4b181a31913d54#diff-20efb33c062ed3c70211a00fb7b62f27f594aabb86039cbac7620de1a551dfa4R83

@asymmetric  and the list of images' names can be found in the https://github.com/nixos/nixpkgs/blob/master/nixos/modules/virtualisation/gce-images.nix

Reply:
@tewfik-ghariani thanks. I was finding them by running `gsutil ls -l gs://nixos-cloud-images`, which interestingly also has a 21.03 image :/